third_party/pyth/multisig-wh-message-builder/src/parser.ts

```diff
@@ -0,0 +1,118 @@
+import { ChainId, CHAIN_ID_SOLANA } from "@certusone/wormhole-sdk";
+import {
+  AccountMeta,
+  PublicKey,
+  TransactionInstruction,
+} from "@solana/web3.js";
+import { pythOracleCoder, pythIdl } from "@pythnetwork/client";
+import { Cluster, CONFIG } from ".";
+import { Idl } from "@project-serum/anchor";
+
+/// A summary of a proposal instruction :
+/// For two proposals A and B in the multisig the set of all the summarized instructions of A and all the summarized instructions of B should have null intersection
+type Summary = {
+  targetChain: ChainId;
+  component: OracleProgram;
+};
+type OracleProgram = {
+  oracleAction:
+    | UpdatePriceFeed
+    | "updatePermissions"
+    | UpdateProductAccount
+    | undefined;
+};
+
+type UpdatePriceFeed = {
+  priceAccount: PublicKey;
+};
+
+type UpdateProductAccount = {
+  productAccount: PublicKey;
+};
+
+type NamedAccounts = Record<string, AccountMeta>;
+
+export function parseProposalInstruction(
+  cluster: Cluster,
+  instruction: TransactionInstruction
+) {
+  switch (instruction.programId.toBase58()) {
+    case CONFIG[cluster].pythOracleSolana.toBase58():
+      const deserializedInstruction = pythOracleCoder().instruction.decode(
+        instruction.data
+      );
+      if (deserializedInstruction) {
+        const namedAccounts = resolveAccountNames(
+          pythIdl as unknown as Idl,
+          deserializedInstruction.name,
+          instruction
+        );
+        const summary: Summary = summarizeOracleInstruction(
+          deserializedInstruction.name,
+          namedAccounts
+        );
+        return { ...deserializedInstruction, namedAccounts, summary };
```

Comment:
what's in deserialized instruction? is this representation generic enough to work for the other kinds of operations that will be stored in the multisig?